### ISSUE TITLE
Programmatic import on AWS?

### ISSUE THREAD

**Issue by iboates** [3dcitydb/importer-exporter#47]
Is there a way to run a database import via command line arguments or some other programmatic method? I would like to deploy an xml file to a cloud server & DB along with the importer/exporter tool and then have the import happen there rather than tie up my own machine.

**Comment by FxKu** [comment]
Yes, the Importer/Exporter also provides a command line interface. You can find a description in the 3DCityDB manual starting from page 202 - chapter 5.8.

**Comment by iboates** [comment]
Thanks for the info. I was able to get it working. I do have one further question, though: Is it also possible to do the installation itself from the command line? I am unable to simply run `java -jar 3DCityDB-Importer-Exporter-3.3.2-Setup.jar` because it will attempt to launch a GUI, which throws an error because the AWS virtual machine has no graphical display, only a command line.

I intend to document the procedure of setting this whole system up in the AWS architecture, would your group be interested in that, maybe even adding it as an addendum to the documentation? I would like to contribute to the project.

**Comment by clausnagel** [comment]
Well, we are using IzPack 4 for creating the installer. IzPack supports different ways for non-GUI installations. Check out the chapters `Unattended installations` and `Console installations` to learn how it works here: https://izpack.atlassian.net/wiki/spaces/IZPACK/pages/491532/Advanced+Features

Another possibility is to install the importer/exporter once using the GUI, and to zip the installation directory afterwards. Then you can unzip this archive at different locations using the console.

Also take a look the 3DCityDB docker images at: https://github.com/tum-gis/3dcitydb-docker-postgis

Yes, we very much welcome contribution to the 3DCityDB project. So please feel encouraged to do so. Adding another addendum to the documentation would definitely be one option.

**Comment by clausnagel** [comment]
Closing due to inactivity. Feel free to re-open if you need further assistance.

**Comment by HuyQuoc** [comment]
> Closing due to inactivity. Feel free to re-open if you need further assistance.

Hi, What is supported version of Ubuntu for installing without GUI ? Is this support 16.xx ? 

Regards.

**Comment by clausnagel** [comment]
Actually, the Importer/Exporter only requires a JDK 8 (or hight) to be properly set up on the machine. There is not dependency on specific versions of Ubuntu. Are you facing issues on Ubuntu 16.x?

**Comment by HuyQuoc** [comment]
> Actually, the Importer/Exporter only requires a JDK 8 (or higher) to be properly set up on the machine. There is no dependency on specific versions of Ubuntu. Are you facing issues on Ubuntu 16.x?

I'm trying to do in Ubuntu 16.x but it's no longer support on servers. So i cannot update components to install Importer/Exporter.

**Comment by clausnagel** [comment]
But Java 8 should be availble for Ubuntu 16?!  Or use [Java 11](http://jdk.java.net/11/) which only requires unzipping to your machine. If this does not work, you should contact the Ubuntu support.

**Comment by HuyQuoc** [comment]
> But Java 8 should be availble for Ubuntu 16?! Or use [Java 11](http://jdk.java.net/11/) which only requires unzipping to your machine. If this does not work, you should contact the Ubuntu support.

Hi, I installed Ubuntu 16.10, so it's no longer support. I re-installed Ubuntu 16.04, but it still can not setup Importer/Exporter. It only return two folders (plugins and uninstaller) after run *.jar file, no more. 

Here is my command: java -jar 3DCityDB-Importer-Exporter.jar -console.

**Comment by clausnagel** [comment]
Maybe an issue with IzPack. I cannot test this currently.

The easiest way to overcome all these issues is to install the Importer/Exporter using the GUI on some machine with GUI support. Then zip or tar the installation directory and unzip on your Ubuntu machine.

**Comment by HuyQuoc** [comment]
> Could you please open a new ticket in the 3dcitydb/3dcitydb repo as this seems to be more database realated. Btw the error messages tell me that postgis is not added to your chosen database. How do you run the setup? Thru the CREATE_DB shell script? Did you change the database name in the CONNECTION_DETAILS.SH file?

Yes, I've changed information in CONNECTION_DETAILS.SH. I tried many times but have same errors and I dont know reason. 

Thanks.

**Comment by clausnagel** [comment]
I found the time to test the IzPack console installation, and was able to reproduce the issue reported from @HuyQuoc. The installation only creates the folders `plugins` (if selected to install) and `uninstall`. This is a bug in IzPack 4.3.5, which we use for building the Importer/Exporter 4.0 installer. 

I fixed the bug (see 537510bd15b1e132ba0b8187ad49546598aa02d3). The console installation now works as expected. The fix will be included in the next release.
